contracts/AllowedStrategy.sol

```diff
@@ -0,0 +1,270 @@
+
+    /**
+     * @notice Used when attempting to allow strategy that is already allowed.
+     * @param group The group's address.
+     */
+    error StrategyAlreadyAdded(address group);
+
+    /**
+     * @notice Used when a strategy does not meet the validator group health requirements.
+     * @param group The group's address.
+     */
+    error StrategyNotEligible(address group);
+
+    /**
+     * @notice Used when attempting to withdraw from allowed strategy
+     * but group does not have enough Celo. Group either doesn't have enough stCelo
+     * or it is necessary to rebalance the group.
+     * @param group The group's address.
+     */
+    error GroupNotBalancedOrNotEnoughStCelo(address group);
+
+    /**
+     * @notice Used when there isn't enough CELO voting for a account's strategy
+     * to fulfill a withdrawal.
+     * @param group The group's address.
+     */
+    error CantWithdrawAccordingToStrategy(address group);
+
+    /**
+     * @notice Used when attempting to deposit when there are not active groups
+     * to vote for.
+     */
+    error NoActiveGroups();
+
+    /**
+     * @notice Initialize the contract with registry and owner.
+     * @param _registry The address of the Celo registry.
+     * @param _owner The address of the contract owner.
+     * @param _manager The address of the Manager contract.
+     */
+    function initialize(
+        address _registry,
+        address _owner,
+        address _manager
+    ) external initializer {
+        _transferOwnership(_owner);
+        __UsingRegistry_init(_registry);
+        __Managed_init(_manager);
+    }
+
+    /**
+     * @notice Set this contract's dependencies in the StakedCelo system.
+     * @param _account The address of the Account contract.
+     * @param _groupHealth The address of the Group health contract.
+     */
+    function setDependencies(address _account, address _groupHealth) external onlyOwner {
+        require(_account != address(0), "account null address");
+        require(_groupHealth != address(0), "validate group null address");
+
+        account = IAccount(_account);
+        groupHealthContract = IGroupHealth(_groupHealth);
+    }
+
+    /**
+     * @notice Marks a group as allowed strategy for voting.
+     * @param group The address of the group to add to the set of allowed
+     * strategies.
+     */
+    function allowStrategy(address group) external onlyManager {
+        if (!groupHealthContract.isValidGroup(group)) {
+            revert StrategyNotEligible(group);
+        }
+
+        if (allowedStrategies.contains(group)) {
+            revert StrategyAlreadyAdded(group);
+        }
+
+        if (!allowedStrategies.add(group)) {
+            revert FailedToAddAllowedStrategy(group);
+        }
+
+        emit StrategyAllowed(group);
+    }
+
+    /**
+     * @notice Marks a group as not allowed strategy for voting
+     * and redistributes votes to default strategy.
```

Comment:
I don't see redistribution of votes happening anywhere in this function?

Reply:
This is a leftover from previous design - thanks